src/commands/force/source/pull.ts

```diff
@@ -113,7 +115,7 @@ export default class Pull extends SourceCommand {
     if (this.flags['api-version']) {
       componentSet.apiVersion = this.flags['api-version'];
     }
-    const username = this.flags['target-org'].getUsername();
+    const username = this.flags['target-org'].getUsername() as string;
```

Comment:
I wonder why `getUsername` says this is optional when the username is always expected: https://github.com/forcedotcom/sfdx-core/blob/b3c1d6b38aed89b063858352a0b9bc845e814a0f/src/org/connection.ts#L77

https://github.com/forcedotcom/sfdx-core/blob/b3c1d6b38aed89b063858352a0b9bc845e814a0f/src/org/org.ts#L844

Reply:
The sfdx-core `Org` class allows you to instantiate an Org (the class), assign various props, and then write it for the first time.

I've always wanted to make either an option on Org.create to say, "read this from the FS and throw" and then have better typing on that version.  Or to have a method that verifies it's coming from the FS and then you can depend on all the expected props being there.

It's kind of hard, though, with a bunch of these being define as class props rather than a type-able object.